src/cloud/functions/leaderboards/around-me/leaderboard/run.csx

```diff
@@ -0,0 +1,240 @@
+    // Remarks: This initialization will run once on every instance and on every recompile of this function
+    if (runOnce)
+    {
+        log.Info("Running initialization");
+
+        if (string.IsNullOrWhiteSpace(baseArchitectureVersion) ||
+            !baseArchitectureVersion.StartsWith(requiredBaseArchitectureVersion)) log.Error($"The base architecture version doesn't match the expected version {requiredBaseArchitectureVersion}");
+
+        // Check required application settings
+        if (string.IsNullOrWhiteSpace(db)) log.Error("COSMOSDB_DBNAME settings wasn't provided");
+        if (string.IsNullOrWhiteSpace(endpoint)) log.Error("COSMOSDB_ENDPOINT settings wasn't provided");
+        if (string.IsNullOrWhiteSpace(key)) log.Error("COSMOSDB_PRIMARY_MASTER_KEY settings wasn't provided");
+
+        // Create Cosmos DB Client from settings
+        client = new DocumentClient(new Uri(endpoint), key);
+
+        // Create Database and Collection in Cosmos DB Account if they don't exist
+        await client.CreateDatabaseIfNotExistsAsync(new Database { Id = db });
+        await client.CreateDocumentCollectionIfNotExistsAsync(
+            UriFactory.CreateDatabaseUri(db),
+            new DocumentCollection { Id = collection });
+
+        runOnce = false;
+
+        log.Info("Initialization done!");
+    }
+
+    if (string.IsNullOrWhiteSpace(leaderboard) || string.IsNullOrWhiteSpace(playerId))
+        return req.CreateResponse(HttpStatusCode.BadRequest, "Both playerId and leaderboard should be provided in querystring, format is leaderboard/playerId");
+
+    try
+    {
+
+        var scores = client.CreateDocumentQuery<ScoreItem>(
+            UriFactory.CreateDocumentCollectionUri(db, collection), new FeedOptions { EnableCrossPartitionQuery = true });
+
+        // Get score for the current player by playerId        
+        var query =
+            (from s in scores
+             where s.PlayerId == playerId && s.Leaderboard == leaderboard
+             select s);
+
+        var player = query.ToList();
+        if (player.Count < 1)
+            return req.CreateResponse(HttpStatusCode.BadRequest, $"Player with Id = {playerId} in Leaderboard = {leaderboard} does not exist in the database");
+
+        if (player.Count > 1)
+            return req.CreateResponse(HttpStatusCode.BadRequest, $"There are more than one player with Id = {playerId} in Leaderboard = {leaderboard}. This is DB confuguration error - this combination should be unique");
+
+        var currentPlayer = player.First();
+
+        // Calculate global rank for the current player by playerId 
+        int globalRank;
+
+        query = from c in scores
+                where c.Score > currentPlayer.Score && c.Leaderboard == leaderboard
+                select c;
+
+        List<LeaderboardItem> sortedResult = new List<LeaderboardItem>(); //Sorted list from highest score (start of list) to lowest for output  
+
+        globalRank = query.Count<ScoreItem>() + 1;
```

Comment:
I'm not 100% sure of this, but as far as I'm aware the System.Linq.Enumerable.Count method will execute on top of an IEnumerable, hence it'll first evaluate the query, transform it to an IEnumerable and then count the number of items. As mentioned above, this will trigger the query to execute and potentially return millions of unnecessary information to be transfered from the database to the function in order to be counted in memory.

I might be incorrect on this but I think we should check it out before we approve this, since this might potentially have a very negative impact on the performance of this leaderboard implementation.

The desired behavior is to have DocumentDB return the count (only the number) instead of retrieving all data that we don't need.

Reply:
To make it safe will transform to IQuearyable
globalRank = (from c in scores
                    where c.Score > currentPlayer.Score && c.Leaderboard == leaderboard
                    select c).Count<ScoreItem>() + 1;